whelk-core/src/main/groovy/whelk/JsonLd.groovy

```diff
@@ -670,6 +704,40 @@ class JsonLd {
         }
     }
 
+    List makeSearchKeyParts(Map object) {
+        Map lensGroups = displayData.get('lensGroups')
+        Map lensGroup = lensGroups.get('chips')
+        Map lens = getLensFor(object, lensGroup)
+        List parts = []
+        def type = object.get(TYPE_KEY)
+        // TODO: a bit too hard-coded...
+        if (type instanceof String && type != 'Identifier' && isSubClassOf(type, 'Identifier')) {
+            parts << type
+        }
+        if (lens) {
+            List propertiesToKeep = (List) lens.get("showProperties")
+            for (prop in propertiesToKeep) {
+                def values = object[prop]
+                if (isLangContainer(context[prop]) && values instanceof Map) {
+                    values = locales.findResult { values[it] }
+                }
+                if (!(values instanceof List)) {
+                    values = values ? [values] : []
+                }
+                // TODO: find recursively (up to a point)? For what? Only
```

Comment:
Should this be fixed later? If so, consider creating a ticket and referencing it here so it isn't forgotten.